saleor/payment/gateways/adyen/plugin.py

```diff
@@ -612,10 +629,27 @@ def confirm_payment(
             # standard flow for confirming an additional action
             return self._process_additional_action(payment_information, kind)
 
-        result_code = transaction.gateway_response.get("resultCode", "").strip().lower()
-        payment_method = (
-            transaction.gateway_response.get("paymentMethod", "").strip().lower()
-        )
+        result_code_temporary_field = transaction.legacy_adyen_plugin_result_code
+        payment_method_temporary_field = transaction.legacy_adyen_plugin_payment_method
+
+        if (not result_code_temporary_field) and (not payment_method_temporary_field):
```

Comment:
Suggestion (non-blocking)

The brackets are useless.
```suggestion
        if not result_code_temporary_field and not payment_method_temporary_field:
```